### ISSUE TITLE
New Rule: `svelte/no-dupe-slot-name` rule

### ISSUE THREAD

**Issue by ota-meshi** [sveltejs/eslint-plugin-svelte#309]
### Motivation

I would like a new rule to warn when using the same slot name multiple times.

```svelte
<ContactCard>
  <span slot="name">
    Foo
  </span>
  <span slot="name">
    Bar
  </span>
</ContactCard>
```

### Description

The same slot name cannot be used multiple times. It gives an error in Svelte's compiler.
If Svelte warns you, you might think you don't need ESLint rule.
I checked it and it seems to be the render (generate) phase that errors out on duplicate slot names. Since the error is not detected at the time of compilation  on language server, there is a possibility that we will notice it late.

So I think we need a rule to check for duplicate slot names.

### Examples

```svelte

<!-- ✓ GOOD -->
<Foo>
  <span></span>
  <span></span>
</Foo>
<Foo>
  <span slot="firstName">John</span>
  <span slot="lastName">Smith</span>
</Foo>
<Foo>
  <span slot="default"></span>
</Foo>

<!-- ✗ BAD -->
<Foo>
  <span></span>
  <span slot="default"></span>
</Foo>
<Foo>
  <span slot="name">John</span>
  <span slot="name">Smith</span>
</Foo>
<Foo>
  text
  <span slot="default"></span>
</Foo>
```

**Comment by baseballyama** [comment]
I got a compile error.

https://svelte.dev/repl/ed634ccfee984a8cab06235cc48ddd06?version=3.53.1

**Comment by ota-meshi** [comment]
You're right, but the problem is that we can't catch warnings when using `compile()` with the `"generate: false"` option and when using `parse()`.

- `compiler.parse(code)` ... can't catch the error.
- `compiler.compile(code, { generate: false })` ... can't catch the error.
- `compiler.compile(code)` ... catch the error.

https://stackblitz.com/edit/sveltejs-kit-template-default-i68vml?file=src/routes/+page.server.js


Language servers and this eslint-plugin-svelte and eslint-plugin-svelte3 use them, so they can't currently catch the problem.
Therefore, the user may notice the problem a little later.
Users will notice the problem when compiling with vite or rollup.
I want the user to be notified of the problem while writing the code.

**Comment by baseballyama** [comment]
Ah, now I got it. Thank you for your explanation.
My next question is that, if a user runs `npm run dev`, immediately the user can see the console error on VSCode right?
So I think that even now, users can know this error while writing the code.

And according to the [docs](https://svelte.dev/docs#template-syntax-slot), there is no concrete spec regarding this error.
So I'm worried that we need to follow such unclear specs in the future.
(For instance, maybe in the future, we should support using `slot` inside `{#if}` statement.)

And at least we need to consider `<svelte:fragment>`.

<img width="910" alt="image" src="https://user-images.githubusercontent.com/19153718/204071504-97bc4a11-6ce3-45d3-a2e2-5fe058ac18ea.png">

**Comment by ota-meshi** [comment]
Thank you for letting me know.
If there is a possibility that the slot name limit will be changed, I think this rule has a very low priority.
I would like to keep this issue open and see how it progresses 😉.

**Comment by baseballyama** [comment]
Svelte5 throws this in compile phase, so it's easy to recognize this.

https://svelte.dev/playground/hello-world?version=5.16.1#H4sIAAAAAAAAE4XNywrCMBCF4VcZzkqh6D5UQQviO1gXsRmhEDMhGW-UvrukICgIbn--OTMg2AvDYM_eC90leUczdr2ym6PCufecYQ4D9BmLKwHV-2oT4yLf2GtpJ5v5V-8kKAfNMKgbCWo7bWxy6zYQ1TnaQNmLrtpptMXUiXYiE1gW8cdubfqy9fLzDyooPxRG05XH4_gCzcLtxPQAAAA=

And `<svelte:fragment>` is deprecated in Svelte5.

So I close this issue.
